TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementCheckers.cs

```diff
@@ -16,10 +18,39 @@ public static class SavepointStatementChecker
                 if (savePointName.StartsWith("SYS", StringComparison.OrdinalIgnoreCase))
                 {
                     DiagnosticUtils.AddError(savepointStatement,
-                        "Invalid savepoint-name, it must not begin with 'SYS'.",
+                        "Invalid savepoint-name, it must not begin with 'SYS'.", context);
+                }
+            }
+        }
+    }
+
+    public static class SetAssignmentStatementChecker
+    {
+        internal static void OnCodeElement(SetAssignmentStatement setAssignmentStatement,
+            CodeElementsParser.SetAssignmentStatementContext context)
+        {
+            foreach (var assignment in setAssignmentStatement.Assignments)
+            {
+                IList<SourceValue> values = new List<SourceValue>();
+                IList<TargetVariable> targets = new List<TargetVariable>();
+                if (assignment.Values != null)
+                {
+                    values = assignment.Values;
+                }
+
+                if (assignment.Targets != null)
+                {
+                    targets = assignment.Targets;
+                }
```

Comment:
Looking at the builder code, the targets and values are always non-null, so this is redundant.